enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/marshalling/RaftMessageEncoder.java

```diff
@@ -53,25 +53,67 @@ protected synchronized void encode( ChannelHandlerContext ctx,
         channel.putInt( message.type().ordinal() );
         memberMarshal.marshal( message.from(), channel );
 
-        if ( message instanceof RaftMessages.Vote.Request )
+        message.visit( new Visitor( marshal, memberMarshal, channel ) );
+    }
+
+    private static class Visitor implements RaftMessages.Visitor<Void, Exception>
```

Comment:
Perhaps a better name than just Visitor. EncodingHandler?

Reply:
The outer class name spaces the visitor, so it's clear that it refers to raft message encoding. I've renamed it to Handler.